audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java

```diff
@@ -190,11 +192,22 @@ void testFailedErrorDeserialisation() throws Exception {
         // WHEN - we write to the input and wait
         runStreamOf(requests);
 
-        // THEN - check an "Error-..." file has been created
-        var actualErrorCount = currentErrorCount.get();
-        assertThat(actualErrorCount)
-                .as("Check exactly 1 'Error' file has been created")
-                .isEqualTo(expectedErrorCount);
+        for (int i = 1; i <= N_RUNS; i++) {
+            var actualErrorCount = currentErrorCount.get();
+
+            // THEN - check an "Error-..." file has been created
+            try {
+                assertThat(actualErrorCount)
+                        .as("Check exactly 1 'Error' file has been created")
+                        .isEqualTo(expectedErrorCount);
+            } catch (AssertionError e) {
+                LOGGER.info("There was an assertion error on attempt {}", i);
+                if (i == N_RUNS) {
+                    throw e;
+                }
+            }
+            TimeUnit.SECONDS.sleep(BACKOFF);
+        }
```

Comment:
Can we stick this in a `tryAssertWithBackoff` method or something?
Pass a `Runnable` as an argument, which here would be `() -> assertThat(currentErrorCount.get()).as(...).isEqualTo(...)`

Reply:
Then if we want more tests here, or want to re-use this piece of code, it's straightforward and we're not repeating ourselves too much